src/util-streaming-buffer.c

```diff
@@ -1204,9 +1224,12 @@ static StreamingBufferRegion *BufferInsertAtRegionConsolidate(StreamingBuffer *s
     return dst;
 }
 
+/** \note sets sc_errno */
 static StreamingBufferRegion *BufferInsertAtRegionDo(StreamingBuffer *sb,
         const StreamingBufferConfig *cfg, const uint64_t offset, const uint32_t len)
 {
+    sc_errno = SC_OK;
```

Comment:
Just a nit depending on how much like `errno` we want to be...  Functions as part of a library never (re)set errno back to 0, that is left up to the caller. Just thinking out loud with respect to a library where we should have documentation on the error handling strategy.

Reply:
I've seen examples where it was set to OK/SUCCESS by the lib call, which to me seems like a more dev friendly pattern. But we can go either way, I guess. Just don't really like having to reset it all over the call sites.

Btw, we're not really resetting it before making syscalls. I guess in an ideal case it won't ever have to be reset, as we'd only check it in case of error, and in case of error it should have been set.